src/utils/caseDefinitionFilters.ts

```diff
@@ -0,0 +1,53 @@
+import { Node } from "../d";
+import { getNodeAttr } from "./treeMethods";
+
+export const filterSamples = (
+  samples: Node[],
+  caseDefFilters: { [key: string]: any },
+  samplesOfInterestNames: string[]
+) => {
+  let matchingSamples: Node[] = samples;
+  if (Object.keys(caseDefFilters).length === 0) {
+    return [];
+  }
+
+  for (let i = 0; i < Object.entries(caseDefFilters).length; i++) {
+    let [field, filter] = Object.entries(caseDefFilters)[i];
+
+    //@ts-ignore
+    if (filter["dataType"] === "categorical") {
+      //@ts-ignore
+      matchingSamples = matchingSamples.filter((n: Node) =>
+        //@ts-ignore
+        filter["acceptedValues"].includes(getNodeAttr(n, field))
+      );
+    } else {
+      //@ts-ignore
+      matchingSamples = matchingSamples.filter(
+        (n: Node) =>
+          //@ts-ignore
+          getNodeAttr(n, field) <= filter["max"] &&
+          getNodeAttr(n, field) >= filter["min"]
+      );
+    }
+  }
+
+  const newMatchingSamples = matchingSamples.filter(
+    (n: Node) => !samplesOfInterestNames.includes(n.name)
+  );
+
+  return newMatchingSamples;
+};
+
+export const fieldIsValid = (fieldSummary: any) => {
+  if (fieldSummary["dataType"] == "categorical") {
+    return (
+      fieldSummary["uniqueValues"].length <= 100 &&
```

Comment:
What happens in the case of categorical data that has a huge number of discrete, non-numerical options? Do those get dropped from the filtering possibilities?

Reply:
Yes